cmd/thanos/sidecar.go

```diff
@@ -153,21 +154,20 @@ func runSidecar(
 	})
 
 	// Setup all the concurrent groups.
+	promAgentModeEnabled := false
 	{
 		promUp := promauto.With(reg).NewGauge(prometheus.GaugeOpts{
 			Name: "thanos_sidecar_prometheus_up",
 			Help: "Boolean indicator whether the sidecar can reach its Prometheus peer.",
 		})
-
 		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
```

Comment:
We probably don't need to defer here, we call `cancel()` in second `interrupt` arg of `g.Add`